crates/sargon-uniffi/src/security_center/support/backup_result.rs

```diff
@@ -0,0 +1,39 @@
+use crate::prelude::*;
+use sargon::BackupResult as InternalBackupResult;
+use sargon::IsBackupResultCurrent;
+use sargon::IsBackupResultFailed;
+
+/// A struct that represents the result of a given backup.
+///
+/// Reference for iOS: it is a combination of `BackupStatus` and `BackupResult` (all in one).
+#[derive(Clone, PartialEq, Eq, uniffi::Record)]
+pub struct BackupResult {
+    /// The identifier of the backup.
+    pub save_identifier: String,
+
+    /// Whether this backup matches the one on Profile.
+    pub is_current: bool,
+
+    /// Whether this backup has failed.
+    pub is_failed: bool,
```

Comment:
what do you define as failed in iOS?
If the cloud backup is off, is it a fail?
Is cloud service error a fail? 
Is it a fail, if cloud backup is on but not updated?
All the above?
What else?

just for reference here is how in android we define the [BackupState](https://github.com/radixdlt/babylon-wallet-android/blob/main/core/src/main/java/rdx/works/core/domain/cloudbackup/BackupState.kt) (cloud + manual)

Reply:
In iOS we store in local storage information about the last manual/cloud backup. This information is updated whenever a new backup is started, and when it finishes (successfully or not). The idea is to create this `BackupResult` based on such information.

> what do you define as failed in iOS?

In iOS we have 3 status for a backup: `success`, `failed` and `started`.
We consider it `is_failed` when the `status == failed` or when `status == started` and it has been in such state for 5 minutes or more ([ref](https://github.com/radixdlt/babylon-wallet-ios/blob/main/RadixWallet/Clients/UserDefaults%2BDependency%2BExtension/UserDefaults%2BDependency%2BExtension.swift#L261-L269)).

> Is cloud service error a fail?
> Is it a fail, if cloud backup is on but not updated?
> All the above?
> What else?

Note that is only refers to the result of the last backup attempt. The considerations to whether it is a problem or not based on cloud backups being disabled/enabled is done [here](https://github.com/radixdlt/sargon/pull/287/files#diff-0703cbaefcade0f2714715250fb3956bda33b19756427ef0c4fc563ee88a98c5R22-R28).

In other words, `BackupResult` is agnostic of the Wallet backup configuration. It only reflects the state of the last backup attempt (either manual or cloud). This was easy to do on iOS because as I mentioned, we save such information on local storage. Is Android keeping track of this information as well?

Reply:
ok in Android it is different. We have two backup states that holds the information that is needed to derive the problems later.

```data class CloudBackupEnabled(
        override val email: String,
        val hasAnyErrors: Boolean = false,  <--------- if true then security problem 5
        override val lastCloudBackupTime: Timestamp? = null,
        override val lastManualBackupTime: Instant? = null,
        override val lastModifiedProfileTime: Timestamp? = null,
    ) : BackupState()

    data class CloudBackupDisabled(
        override val email: String?,
        override val lastCloudBackupTime: Timestamp?,
        override val lastManualBackupTime: Instant?,
        override val lastModifiedProfileTime: Timestamp?
    ) : BackupState()
```

but based on the way you implemented the consideration whether it is a problem or not I think it should be doable to "map" the `BackupState` to `BackupResult`